### PR TITLE
CORE-13895 Ledger group parameters cache

### PR DESCRIPTION
**Pull request by lankydan**
Cache the results of persisting and finding group parameters within the ledger code.

The writeup of the performance impact can be read here - https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4498358273/Test+4+-+Reduce+time+spent+fetching+and+persisting+group+parameters+within+ledger+flows

### CODE REVIEW

**Comment by lankydan** [comment]
This PR depends on https://github.com/corda/corda-runtime-os/pull/3969 hence the signing keys cache change that is irrelevant to this PR.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.osgi.service.component.annotations.Reference
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+@Component(service = [GroupParametersCache::class])
+class GroupParametersCacheImpl @Activate constructor(
+    @Reference(service = CurrentSandboxGroupContext::class)
+    private val currentSandboxGroupContext: CurrentSandboxGroupContext,
+    @Reference(service = SandboxGroupContextComponent::class)
+    private val sandboxGroupContextComponent: SandboxGroupContextComponent
```
Comment (chrisr3) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt]: You should only need to inject a `CacheEviction` reference here.

**Review by chrisr3 — COMMENTED** [review]
Another one?

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt (hunk 1 above)
Comment (lankydan) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt]: Will do, I haven't made the same updates to this PR from the other cache pr yet.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    @Reference(service = CurrentSandboxGroupContext::class)
+    private val currentSandboxGroupContext: CurrentSandboxGroupContext,
+    @Reference(service = CacheEviction::class)
+    private val cacheEviction: CacheEviction
+) : GroupParametersCache, SandboxedCache {
+
+    // TODO Access configuration to setup the cache
+    private companion object {
+        private val log: Logger = LoggerFactory.getLogger(GroupParametersCacheImpl::class.java)
+        private const val MY_SIGNING_KEYS_CACHE_MAX_SIZE_PROPERTY_NAME =
```
Comment (blsemo) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt]: Wrong name? Nothing to do with signing keys.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            "net.corda.ledger.utxo.flow.impl.persistence.group.parameters.cache.maximumSize"
+    }
+
+    private val maximumSize = java.lang.Long.getLong(MY_SIGNING_KEYS_CACHE_MAX_SIZE_PROPERTY_NAME, 10000)
```
Comment (blsemo) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/GroupParametersCacheImpl.kt]: Isn't 10000 group parameters a bit large for a default here? I don't think they're that small, and would you ever have that many?

**Review by blsemo — COMMENTED** [review]
Two small niggles.

**Review by ronanbrowne — APPROVED** [review]
LGTM (Build changes)

**Pull request merged**